webapp/channels/src/components/admin_console/content_flagging/user_multiselector/user_multiselector.tsx

```diff
@@ -53,65 +63,216 @@ type Props = MultiSelectProps & SingleSelectProps & {
-    const selectInitialValue = initialUsers.
-        filter((userProfile) => Boolean(userProfile)).
-        map((userProfile: UserProfile) => ({
-            value: userProfile.id,
-            label: userProfile.username,
-            raw: userProfile,
-        } as AutocompleteOptionType<UserProfile>));
+    const allGroups = useSelector(getAllGroups);
+
+    const selectInitialValue = useMemo(() => {
+        const result: Array<AutocompleteOptionType<UserProfile | Group>> = [];
+        const addedIds = new Set<string>();
+        
+        if (!initialValue) {
+            return result;
+        }
+        
+        // Build a map of user IDs for quick lookup
+        const userMap = new Map<string, UserProfile>();
+        initialUsers.filter(Boolean).forEach((user: UserProfile) => {
+            userMap.set(user.id, user);
+        });
+        
+        // Iterate through initialValue once and add each ID as either user or group
+        initialValue.forEach((id) => {
+            if (addedIds.has(id)) {
+                return; // Skip duplicates
+            }
+            
+            // Try to add as user first
+            const user = userMap.get(id);
+            if (user) {
+                result.push({
+                    value: user.id,
+                    label: user.username,
+                    raw: user,
+                });
+                addedIds.add(id);
+                return;
+            }
+            
+            // If not a user and groups are enabled, try to add as group
+            if (isGroupsEnabled) {
+                const group = allGroups[id];
+                if (group) {
+                    result.push({
+                        value: group.id,
+                        label: group.display_name || group.name,
+                        raw: group,
+                    });
+                    addedIds.add(id);
+                }
+            }
+        });
+        
+        return result;
+    }, [initialUsers, allGroups, initialValue, isGroupsEnabled]);
 
     const userLoadingMessage = useCallback(() => formatMessage({id: 'admin.userMultiSelector.loading', defaultMessage: 'Loading users'}), [formatMessage]);
     const noUsersMessage = useCallback(() => formatMessage({id: 'admin.userMultiSelector.noUsers', defaultMessage: 'No users found'}), [formatMessage]);
     const defaultPlaceholder = formatMessage({id: 'admin.userMultiSelector.placeholder', defaultMessage: 'Start typing to search for users...'});
 
     const generalSearchUsers = useMemo(() => debounce(async (searchTerm: string, callback) => {
         try {
-            const response = await dispatch(searchProfiles(searchTerm, {page: 0}));
-            if (response && response.data && response.data.length > 0) {
-                const users = response.data.
-                    filter((userProfile) => !userProfile.is_bot).
+            const userSearchOptions = {
```

Comment:
Previously, no such filter was being passed to `searchProfiles`, and now we are passing a team ID to filter by. The only reason this still works for content flagging section is because there is no team ID in system console. I don't know if there is a better solution but I'd prefer if this was more explicit.

Reply:
Actually, right now you are only using this component in system console and so am I. Since there is no concept of teams and current team there, why are we even adding this team_id filter?